torch/csrc/distributed/rpc/init.cpp

```diff
@@ -89,10 +91,15 @@ PyObject* rpc_init(PyObject* /* unused */) {
 
   shared_ptr_class_<ProcessGroupAgent>(module, "ProcessGroupAgent", rpcAgent)
       .def(
-          py::init<std::string, std::shared_ptr<::c10d::ProcessGroup>, int>(),
+          py::init<
+              std::string,
+              std::shared_ptr<::c10d::ProcessGroup>,
+              int,
+              std::chrono::milliseconds>(),
           py::arg("name"),
           py::arg("process_group"),
-          py::arg("num_send_recv_threads") = 4)
+          py::arg("num_send_recv_threads") = kDefaultNumSendRecvThreads,
```

Comment:
Can we define a struct, `RpcOptionos`? I am afraid this argument list will keeping growing.

Reply:
Created https://github.com/pytorch/pytorch/issues/28940 to track this. Will do it in a follow up diff.

Reply:
Since they're kwargs, does it matter? Having the defaults be defined inline is actually kinda nice.

Reply:
I would try to reduce the use of kwargs, because that will be a blackbox for users. I prefer a data structure with clearly available field names.